# --- T2-COPYRIGHT-NOTE-BEGIN ---
# This copyright note is auto-generated by ./scripts/Create-CopyPatch.
# 
# T2 SDE: package/.../binutils/gcc-4.0.patch
# Copyright (C) 2004 - 2005 The T2 SDE Project
# 
# More information can be found in the files COPYING and README.
# 
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License as published by the Free Software
# Foundation; either version 2 of the License, or (at your option) any later
# version.
# --- T2-COPYRIGHT-NOTE-END ---

Although we do not have gcc-4.0 in 2.1 we want to be able to bootstrap the
stable series on newer systems. We do not just update binutils, because
we do not want such a "major" change.

  - Rene Rebe <rene@exactcode.de>

--- binutils-2.15.94.0.2.2/gas/tc.h.vanilla	2005-12-15 17:14:40.000000000 +0100
+++ binutils-2.15.94.0.2.2/gas/tc.h	2005-12-15 17:15:17.000000000 +0100
@@ -27,22 +27,6 @@
 /* JF moved this here from as.h under the theory that nobody except MACHINE.c
    and write.c care about it anyway.  */
 
-struct relax_type
-{
-  /* Forward reach. Signed number. > 0.  */
-  long rlx_forward;
-  /* Backward reach. Signed number. < 0.  */
-  long rlx_backward;
-
-  /* Bytes length of this address.  */
-  unsigned char rlx_length;
-
-  /* Next longer relax-state.  0 means there is no 'next' relax-state.  */
-  relax_substateT rlx_more;
-};
-
-typedef struct relax_type relax_typeS;
-
 extern const int md_reloc_size;	/* Size of a relocation record.  */
 
 char * md_atof (int, char *, int *);
--- binutils-2.15.94.0.2.2/gas/as.h.vanilla	2005-12-15 17:14:56.000000000 +0100
+++ binutils-2.15.94.0.2.2/gas/as.h	2005-12-15 17:17:03.000000000 +0100
@@ -605,9 +605,25 @@
 struct expressionS;
 struct fix;
 typedef struct symbol symbolS;
-struct relax_type;
 typedef struct frag fragS;
 
+struct relax_type
+{
+  /* Forward reach. Signed number. > 0.  */
+  long rlx_forward;
+  /* Backward reach. Signed number. < 0.  */
+  long rlx_backward;
+
+  /* Bytes length of this address.  */
+  unsigned char rlx_length;
+
+  /* Next longer relax-state.  0 means there is no 'next' relax-state.  */
+  relax_substateT rlx_more;
+} relax_type;
+
+typedef struct relax_type relax_typeS;
+
+
 #ifdef BFD_ASSEMBLER
 /* literal.c */
 valueT add_to_literal_pool (symbolS *, valueT, segT, int);
